components/raftstore-v2/src/operation/command/admin/compact_log.rs

```diff
@@ -476,8 +495,9 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
             .unwrap();
         self.set_has_extra_write();
 
-        // All logs < perssited_apply will be deleted, so should check with +1.
-        if old_truncated + 1 < self.storage().apply_trace().persisted_apply_index()
+        // All logs < persisted_apply will be deleted.
+        let prev_first_index = first_index;
+        if prev_first_index < self.storage().apply_trace().persisted_apply_index()
```

Comment:
I have a bit worry on this change that it may lead to too many entries cached when evict leader happens. 
Because in Multi-RocksDB, the persisted apply index is updated only after the region's memtable is flushed.  So it can be much bigger entry count than single rocksdb.  It would possibly lead to append message reject because of too many memory usage by raft entry cache. 
I'm wondering if it's better to add the cap of the entries to be cached.

Reply:
I have filed an issue for tracking the enhancement #14884. Will implement it in a later PR.